ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/CustodyGroupCountManagerImpl.java

```diff
@@ -148,6 +156,42 @@ public void setCustodyGroupSyncedCount(final int custodyGroupSyncedCount) {
     custodyGroupSyncedCountGauge.set(custodyGroupSyncedCount);
   }
 
+  private void checkAndHandleGenesisTransition(final Set<UInt64> nodeValidatorIndices) {
+    combinedChainDataClient
+        .getLatestFinalized()
+        .ifPresent(
+            finalizedAnchor -> {
+              // Check if we've moved beyond genesis epoch (epoch 0)
+              if (finalizedAnchor.getEpoch().isGreaterThan(UInt64.ZERO)) {
```

Comment:
In the PR description you said you don't have to wait until epoch 0 is finalized but the whole update cgc count only runs once we have an epoch greater than 0 in this case which means we've already finalized genesis